src/realm/object-store/c_api/query.cpp

```diff
@@ -224,16 +224,15 @@ RLM_API const char* realm_query_get_description(realm_query_t* query)
     });
 }
 
-RLM_API realm_query_t* realm_query_append_query(const realm_query_t* existing_query, const char* query_string,
+RLM_API realm_query_t* realm_query_append_query(realm_query_t* existing_query, const char* query_string,
                                                 size_t num_args, const realm_query_arg_t* args)
 {
     return wrap_err([&]() {
         auto realm = existing_query->weak_realm.lock();
         auto table = existing_query->query.get_table();
         Query query = parse_and_apply_query(realm, table, query_string, num_args, args);
-        Query combined = Query(existing_query->query).and_query(query);
+        Query combined = existing_query->get_query().and_query(query);
         auto ordering_copy = util::make_bind<DescriptorOrdering>();
-        *ordering_copy = existing_query->get_ordering();
```

Comment:
This is a breaking change, the ordering of the existing query will be ignored, all the queries that rely on the existing ordering to be considered are not going to work any more. 
As per this test: 

![image](https://user-images.githubusercontent.com/1497069/189413835-f2c73cfa-3fb7-4e9a-89ae-690fddbe7417.png)


Do we want this to start ignoring? Probably we should append the order if the current query does not have one.

Reply:
This does not apply any more, since code has been reverted.